reactive/swarm.py

```diff
@@ -69,13 +91,108 @@ def swarm_relation_broken():
+    if not is_leader():
+        return
+    else:
+        status_set('maintenance', 'Reconfiguring SSL PKI configuration')
+
+    log('Updating EasyRSA3 OpenSSL Config')
+    openssl_config = 'easy-rsa/easyrsa3/x509-types/server'
+
+    with open(openssl_config, 'r') as f:
+        existing_template = f.readlines()
+
+    # use list comprehension to enable clients,server usage for certificates
+    # with the docker/swarm daemons.
+    xtype = [w.replace('serverAuth', 'serverAuth, clientAuth') for w in existing_template]  # noqa
+    with open (openssl_config, 'w+') as f:
+        f.writelines(xtype)
+
+    reactive.set_state('swarm.tls.opensslconfig.modified')
+    reactive.set_state('easyrsa configured')
+
+
+@when('tls.server.certificate available')
+def enable_client_tls():
+    '''
+    Copy the TLS certificates in place and generate mount points for the swarm
+    manager to mount the certs. This enables client-side TLS security on the
+    TCP service.
+    '''
+    if not path.exists('/etc/docker'):
+        makedirs('/etc/docker')
+
+    kv = unitdata.kv()
+    cert = kv.get('tls.server.certificate')
+    with open('/etc/docker/server.pem', 'w+') as f:
+        f.write(cert)
+    with open('/etc/docker/ca.pem', 'w+') as f:
+        f.write(leader_get('certificate_authority'))
+
+    # schenanigans
+    keypath = 'easy-rsa/easyrsa3/pki/private/{}.key'
+    server = getenv('JUJU_UNIT_NAME').replace('/', '_')
+    if path.exists(keypath.format(server)):
+        copyfile(keypath.format(server), '/etc/docker/server-key.pem')
+    else:
+        copyfile(keypath.format(unit_get('public-address')), '/etc/docker/server-key.pem')
+
+    opts = DockerOpts()
+    config_dir = '/etc/docker'
+    cert_path = '{}/server.pem'.format(config_dir)
+    ca_path = '{}/ca.pem'.format(config_dir)
+    key_path = '{}/server-key.pem'.format(config_dir)
+    opts.add('tlscert', cert_path)
+    opts.add('tlscacert', ca_path)
+    opts.add('tlskey', key_path)
+    opts.add('tlsverify', None)
+    render('docker.defaults', '/etc/default/docker', {'opts': opts.to_s()})
+
+
+@when('swarm.available')
+@when_not('client.credentials.placed')
+def prepare_package():
+    """ Generate a downloadable package for clients to use to speak to the
+    swarm cluster. """
+    if is_leader():
+        client_cert('./swarm_credentials')
+        ca('./swarm_credentials')
+
+        # Prepare the workspace
+        with chdir('./swarm_credentials'):
+            rename('client.key', 'key.pem')
+            rename('client.crt', 'cert.pem')
+            rename('ca.crt', 'ca.pem')
+
+        template_vars = { 'public_address': unit_get('public-address')}
+
+        render('enable.sh', './swarm_credentials/enable.sh', template_vars)
+
+        cmd = 'tar cvf swarm_credentials.tar swarm_credentials'
+        subprocess.check_call(split(cmd))
+        copyfile('swarm_credentials.tar', '/home/ubuntu/swarm_credentials.tar')
+        reactive.set_state('client.credentials.placed')
```

Comment:
The state `client.credentials.placed` is pretty non-specific. It is convention to use the layer name in the state (Yes I know I violated this in the tls layer, and should fix that too).